ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/fillers/CoreProblemFiller.java

```
@@ -86,4 +107,28 @@ private void updateCnecConstraintWithRangeAction(Cnec cnec, RangeAction rangeAct
                         networkElement.getId(),
                         linearRaoData.getSensitivity(cnec, rangeAction)));
     }
+
+    private double getRemedialActionResultValue(RemedialActionElementResult remedialActionElementResult) {
+        if (remedialActionElementResult instanceof PstElementResult) {
+            PstElementResult pstElementResult = (PstElementResult) remedialActionElementResult;
+            return pstElementResult.getPostOptimisationAngle() - pstElementResult.getPreOptimisationAngle();
+        } else if (remedialActionElementResult instanceof RedispatchElementResult) {
+            RedispatchElementResult redispatchElementResult = (RedispatchElementResult) remedialActionElementResult;
+            return redispatchElementResult.getPostOptimisationTargetP() - redispatchElementResult.getPreOptimisationTargetP();
+        }
+        return 0;
```

Please return an error instead.

---

ok